docs/proposals/status.md

```diff
@@ -0,0 +1,300 @@
+
+#### Standard Conditions
+These conditions should be added to `.status.conditions` for Application CRD and subsequently to other resources:
+##### `Ready`
+Readiness as perceived by the resource controller. This means the controller deems the underlying resources to be ready for consumption to the best of its knowledge. It coule be implemented as a low pass filter to filter out transient or persistent failures such as restarts of underlying pods.
+
+##### `Settled`
+Would indicate whether controller acted on latest revision of `.spec` and the underlying resources conform to the latest revision of the `.spec`
+
+##### `Error`
+Would capture Errors seen in last reconcilation loop.
+
+#### Other Conditions from core controllers
+These are not as universal for all kubernetes objects as `Ready`, `Settled`, `Error` and provide yet another dimension of `Ready` and `Settled` condition. 
+
+##### `Progressing`
+Deployment uses this to indicate there is an update to the underlying ReplicaSet that is being acted upon. This is a inversion of the `Settled` condition.
+
+##### `Available`
+Deployment uses this to indicate that underlying pods have been `Ready` for a preconfigured amount of time. This is not a strong heuristic for actual availability but rather a low pass filter that fails when pods are restarting continiously.
+
+#### Other Conditions
+##### `Disruption`
+`Disruption` is a special case of `Error` where we may know what caused the error when `Ready` is false. It is useful for some resources and could be standardized as an optional condition. 
+
+### Aggregating status
+#### Custom Resources
+For custom resources, look for `Ready`, `Error` and `Settled`. If the object does not have the necessary conditions mark that objects status as "Unknown" and would not be used for computing the aggregate status. This would cover the case for nested Applciation CRDs as well. Based on feedback from users and reviewers this stance could be updated.
+
+### Core controller resources
+The logic to compute the readiness for the objects is based on the resource group-kind.
+1. Statefulset is Ready if `.status.readyreplicas == .spec.replicas && .status.currentreplicas == .spec.replicas`
+2. Service is Ready always (could not derive from `.status.loadbalancer`)
```

Comment:
You mean, it will be hard to convince the various implementors of the `Service` controller to implement a status condition that says something like "this `Service` is fully initialized?" To me it seems like that is the point of having standard conditions, no?

Reply:
It is. I meant to say it is outside the scope of this proposal. But we can start pushing in that direction after doing it in the core controllers.